core/sql/executor/ExHbaseIUD.cpp

```diff
@@ -1042,7 +1104,10 @@ ExWorkProcRetcode ExHbaseAccessUpsertVsbbSQTcb::work()
 
 		numRetries_++;
 		return WORK_CALL_AGAIN;
-	      }
+                
+                step_ = PROCESS_INSERT_AND_CLOSE;
```

Comment:
I think a bug might have crept in here. This statement can't be reached because of the return at line 1106.